apps/utils/cipher/cipher.main.kts

```diff
@@ -0,0 +1,505 @@
+                i += 1
+                key = args[i]
+            }
+
+            "--endpoint" -> {
+                i += 1
+                endpoint = args[i]
+            }
+
+            "--region" -> {
+                i += 1
+                region = args[i]
+            }
+
+            else -> {
+                throw UsageException("invalid option ${args[i]}")
+            }
+        }
+        i += 1
+    }
+
+    if (key == null) {
+        throw UsageException("required --key or CIPHER_KEY not provided")
+    }
+
+    val remaining = args.size - i
+    if (remaining < 2 || remaining > 3) {
+        throw UsageException("expected 2 or 3 arguments")
+    }
+
+    val mode = args[i++]
+    val input = args[i++]
+    val output = if (i < args.size) args[i] else input
+
+    return Config(key, endpoint, region, mode, input, output, editor)
+}
+
+fun main(args: Array<String>) {
+    val config = parseArgs(args)
+    val cipher = KmsCipher(config.endpoint, config.region, config.key)
+    val editor = config.editor
+
+    val mode = config.mode
+    val inputFile = File(config.input)
+    val outputFile = File(config.output)
+    when (mode) {
+        "cat" -> {
+            val original = loadText(inputFile)
+            val decrypted = original.decrypt(cipher)
+            val printBuffer = Buffer()
+            decrypted.store(printBuffer)
+            print(printBuffer.readUtf8())
+        }
+
+        "encrypt" -> {
+            val original = loadText(inputFile)
+            val encrypted = original.encrypt(cipher)
+            encrypted.store(outputFile.sink())
+        }
+
+        "decrypt" -> {
+            val original = loadText(inputFile)
+            val decrypted = original.decrypt(cipher)
+            decrypted.store(outputFile.sink())
+        }
+
+        "rewind" -> {
+            val original = loadText(inputFile)
+            val rewound = original.rewind(cipher)
+            rewound.store(outputFile.sink())
+        }
+
+        "edit" -> {
+            val currentDirectory = Paths.get(".")
+            val permissions = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-------"))
+            val extension = if (inputFile.extension.length > 0) ".${inputFile.extension}" else ".txt"
+            val tempFile = Files.createTempFile(currentDirectory, "edit-", extension, permissions).toFile()
+            tempFile.deleteOnExit()
+            try {
+                val original = loadText(inputFile)
+                val cipherCache = original.extractCipherCache(cipher)
+                val rewound = original.rewind(cipher)
+                rewound.store(tempFile.sink())
+                ProcessBuilder(listOf(editor, tempFile.path))
+                    .redirectInput(ProcessBuilder.Redirect.INHERIT)
+                    .redirectOutput(ProcessBuilder.Redirect.INHERIT)
+                    .redirectError(ProcessBuilder.Redirect.INHERIT)
+                    .start()
+                    .waitFor()
```

Comment:
nit: My `EDITOR` is set to `code -w` ([`-w`/`--wait`](https://code.visualstudio.com/docs/editor/command-line#_core-cli-options) instructs VS Code wait until it's closed before returning to its parent process, necessary for scripts that expect that opening the `EDITOR` will pause execution until it's closed), so this script attempts to open an executable named `code -w`, instead of `code` with the `-w` flag. 

_Definitely_ not a big deal, as I can just specify `EDITOR=nvim kotlin ...` as a workaround, but I figured I'd leave this comment for posterity.

Reply:
OK that's a great feature for the next sprint.  Maybe split the editor string on spaces.  I'd hate to get too elaborate with it.  It could get really elaborate if we wind up dealing with quoted strings etc but simple spaces should be fine.